core/src/sumeragi/mod.rs

```diff
@@ -202,24 +202,27 @@ impl SumeragiHandle {
 
     /// Deposit a sumeragi network message.
     pub fn incoming_message(&self, msg: MessagePacket) {
-        if msg.message.is_none() {
-            if let Err(error) = self.control_message_sender.try_send(ControlFlowMessage {
-                view_change_proofs: msg.view_change_proofs,
-            }) {
-                self.metrics.dropped_messages.inc();
-                error!(
-                    ?error,
-                    "This peer is faulty. \
-                     Incoming control messages have to be dropped due to low processing speed."
-                );
+        match msg {
+            MessagePacket::ControlFlowMessage(msg) => {
+                if let Err(error) = self.control_message_sender.try_send(msg) {
+                    self.metrics.dropped_messages.inc();
+                    error!(
+                        ?error,
+                        "This peer is faulty. \
+                         Incoming control messages have to be dropped due to low processing speed."
```

Comment:
Not for this PR, but what do you think about instead of dropping new messages we would overwrite old ones?

Reply:
I think that would be worse from a security perspective.